hibernate-core/src/main/java/org/hibernate/sql/ast/spi/AbstractSqlAstTranslator.java

```diff
@@ -6912,69 +6886,58 @@ else if ( bindValueCount < bindValuePaddingCount ) {
 		String separator = NO_SEPARATOR;
 
 		final Iterator<Expression> iterator = listExpressions.iterator();
-		int itemNumber = 0;
-		while ( iterator.hasNext() && ( inExprLimit == 0 || itemNumber < inExprLimit ) ) {
-			final Expression listExpression = itemAccessor.apply( iterator.next() );
+		Expression listExpression = null;
+		for ( int i = 0; i < bindValueCountWithPadding; i++ ) {
+			if ( inExprLimit > 0 && i % inExprLimit == 0 && i != 0 ) {
+				appendInClauseSeperator( inListPredicate );
```

Comment:
Here a simple rewrite that allows avoiding the costly modulo operation.

```suggestion
		Expression listExpression = null;
		int clauseItemNumber = 0;
		for ( int i = 0; i < bindValueCountWithPadding; i++, clauseItemNumber++ ) {
			if ( inExprLimit > 0 && inExprLimit == clauseItemNumber ) {
				clauseItemNumber = 0;
				appendInClauseSeperator( inListPredicate );
```

Reply:
Looks great. I just took this as is.